lms/djangoapps/courseware/tests/test_credit_requirements.py

```diff
@@ -133,6 +138,29 @@ def test_credit_requirements_not_eligible(self):
         )
         self.assertContains(response, "Verification Failed")
 
+    @ddt.data(CourseMode.HONOR, CourseMode.AUDIT, CourseMode.PROFESSIONAL)
+    def test_credit_requirements_non_credit_enrollment(self, enrollment_mode):
+        # Test the progress table is not displayed to the non credit students.
+        self.enrollment.mode = enrollment_mode
+        self.enrollment.save()  # pylint: disable=no-member
+
+        # Check the progress page display
+        response = self._get_progress_page()
+        self.assertNotContains(response, self.REQUIREMENT_HEADING)
```

Comment:
As @jimabramson suggested, I'd encourage you to check for the presence of the `credit-eligibility` section. Doing so would make these tests more robust to text changes in the template. You also don't need to use a constant for this value, since it's only used in a single place.